lib/kubeclient.rb

```diff
@@ -86,8 +86,8 @@ def get_entities(entity_type, klass, options)
     def watch_entities(entity_type, resource_version = nil)
       resource_name = get_resource_name(entity_type.to_s)
 
-      uri = api_endpoint.merge(
-        File.join(api_endpoint.path, 'watch', resource_name))
+      uri = @api_endpoint.merge(@api_endpoint.path + '/' + @api_version + '/' \
+                                + 'watch' + '/' + resource_name)
```

Comment:
Concatenating `'/' + 'watch' + '/'` is unnecessary. You can just do:

    uri = @api_endpoint.merge(@api_endpoint.path + '/' + @api_version +
                              '/watch/' + resource_name)

(Same for '\' as before, it shouldn't be required)